client.go

```diff
@@ -163,14 +164,15 @@ func (c *client) Scan(s *hrpc.Scan) ([]*hrpc.Result, error) {
 		if err != nil {
 			return nil, err
 		}
+
 		scanres = res.(*pb.ScanResponse)
 		results = append(results, scanres.Results...)
 
 		// TODO: The more_results field of the ScanResponse object was always
 		// true, so we should figure out if there's a better way to know when
 		// to move on to the next region than making an extra request and
 		// seeing if there were no results
-		for len(scanres.Results) != 0 {
+		for len(scanres.Results) != 0 && uint32(len(results)) < numberOfRows {
```

Comment:
My bad, I'm realizing now that` numberOfRows` option is meant to optimize how many rows at max should be returned in each ScanResponse (for each RPC), and it's not meant to limit number of rows returned for the global Scan: http://opentsdb.github.io/asynchbase/javadoc/index.html. Otherwise, when we do scan between 2 keys, we always will have to know how many rows there are between them. What do you think?